### PR TITLE
Add the basic infrastructure and separation between game and app

### PR DESCRIPTION
**Pull request by luanpotter**
This will allows us to work together w/o conflicts

### CODE REVIEW

File: lib/game/player_ship.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import 'package:flame/components.dart';
+import 'package:flame/palette.dart';
+
+class PlayerShip extends PositionComponent {
+  static final _paint = BasicPalette.magenta.paint();
+
+  PlayerShip() {
+    this.anchor = Anchor.center;
+    this.size = Vector2.all(24);
```
Comment (erickzanardo) [inline lib/game/player_ship.dart:11]:
```suggestion
    anchor = Anchor.center;
    size = Vector2.all(24);
```

File: lib/game/player_ship.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    c.drawRect(size.toRect(), _paint);
+  }
+
+  @override
+  void onGameResize(Vector2 gameSize) {
+    super.onGameResize(gameSize);
+
+    this.x = gameSize.x / 2;
+    this.y = gameSize.y - 32;
```
Comment (erickzanardo) [inline lib/game/player_ship.dart:26]:
```suggestion
    x = gameSize.x / 2;
    y = gameSize.y - 32;
```

**Review by erickzanardo — APPROVED** [review]
Couple of nits but LGTM already

**Comment by luanpotter** [comment]
Go home, github, you are drunk haha

**Pull request closed without merging**